### PR TITLE
feat: made newsletter form accessible 🚀

### LINKED ISSUE #511: FEAT: MAKE NEWSLETTER FORM ACCESSIBLE 💡
### Is your feature request related to a problem? Please describe.

- This feature related to problems with accessibility standards.

### Describe the solution you'd like

- I would like to improve the newsletter form accessibility on the homepage by adding important label element with screen reader only class and associated styles for it.
- So it could perform best on accessibility cases.

### Describe alternatives you've considered

- There's no alternative for labels in terms of accessibility

## Screenshots (Going to make changes on these elements)
Screenshot 2023-07-22 at 20-19-40 Fashion Forge 3D

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSoC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by 0xabdulkhalid**
## Related Issue
Close #511

### Describe the changes you've made
- Added proper `label` for mail input on newsletter form.
- The `label` is properly linked to their corresponding `input` element by utilizing the `for`, `name` and `id` attributes.
- The `label` elements can't be seen via the user because they are all hidden by utilizing `sr-only` class, where the screen readers only know them.
- So our newsletter form is much more accessible than before and ensures the accessibility principles.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Tested using NVDA Screen-reader

## Checklist

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)
- We can't take screenshots of accessibility changes rather than code changes, Accessibility changes are often invisible to the user because it's intended to guide disabled people.
## Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/amanjaiman1/Product_3D/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by 0xabdulkhalid** [comment]
Hey there @amanjaiman1, @balaji-sivasakthi 👋🏻

Please make sure to review this PR.

Thank you!

File: src/containers/NewsLetter/index.jsx [hunk 1]
```diff
@@ -53,8 +53,11 @@ function NewsLetter() {
         </motion.div>
 
         <form className="flex flex-col md:flex-row pt-3 md:pt-5 ml:w-full md:w-full lg:w-[600px]">
+          <label for="email" class="sr-only-label">Please provide your email address to subscribe to our newsletter</label>
```
Comment (balaji-sivasakthi) [inline src/containers/NewsLetter/index.jsx:56]: This is not a right method to implement `labels `in ReactJS. Please check how to do it properly.
- `class `to `className`
- `for `to for `htmlFor`

File: src/index.css [hunk 2]
```diff
@@ -43,6 +43,21 @@ body {
   display: none;
 }
 
+/*
+ Screen reader only label, hidden for normal users
+*/
+
+.sr-only-label {
```
Comment (balaji-sivasakthi) [inline src/index.css:50]: Try to implement this style in `tailwind css`

Comment (0xabdulkhalid) [reply src/index.css]: Done!

File: src/containers/NewsLetter/index.jsx (hunk 1 above)
Comment (0xabdulkhalid) [reply src/containers/NewsLetter/index.jsx]: Done!

**Comment by 0xabdulkhalid** [comment]
@balaji-sivasakthi is now everything right in this PR ?

**Comment by balaji-sivasakthi** [comment]
@0xabdulkhalid Done 🎉

**Comment by balaji-sivasakthi** [comment]
@amanjaiman1 merge it

**Comment by balaji-sivasakthi** [comment]
Created new PR for development branch; temporarily locked main branch for maintenance.

**Pull request closed without merging**